src/io.ts

```diff
@@ -91,15 +91,16 @@ export function readTodoStorageFile(todoStorageFilePath: string): Operation[] {
     encoding: 'utf8',
   });
 
-  let operations = todoContents.split(EOL) as OperationOrConflictLine[];
+  // when splitting by EOL, make sure to filter off the '' caused by the final EOL
+  let operations: OperationOrConflictLine[] = todoContents.split(EOL).filter((op: OperationOrConflictLine | '') => op !== '')
 
   if (hasConflicts(todoContents)) {
     operations = resolveConflicts(operations);
 
     writeTodoStorageFile(todoStorageFilePath, operations as Operation[]);
   }
 
-  return operations.filter(Boolean) as Operation[];
+  return operations as Operation[];
```

Comment:
Because empty strings are no longer in this array, this no longer needs the `filter(Boolean)`. This function is should be no different than before to external callers